docs/design/properties_code_point_trie.md

```diff
@@ -0,0 +1,55 @@
+
+`CodePointTrie` code in ICU4C is implemented with a mutable builder, a method to convert the mutable builder to an immutable version, and code to read from the immutable version. The immutable version is stored in memory the same as it is serialized to persistent storage.
+
+## Implementing `CodePointTrie` in ICU4X 
+
+See previous discussion in [issue 131](https://github.com/unicode-org/icu4x/issues/131) regarding BytesTrie and `CodePointTrie`, and the [full meeting minutes](https://docs.google.com/document/d/1oaFovJiRbuBG-O9aq0h69gpGxMH0a_pHmMkO5YweS0E/edit#) of the related meeting for further details.
+
+### Options
+
+#### Option 1: Re-implement `CodePointTrie` in Rust in ICU4X
+
+This option entails writing new Rust code that supports all of the functionality of the `CodePointTrie`.
+
+The advantages would be to have a Rust-native code that can compile entirely within ICU4X and potentially be used elsewhere.
+
+The disadvantages would be the large amount of time to implement all of the complicated logic (largely located in the builder), and the possibility of losing feature parity 'sync' if/when ICU makes further improvements to the `CodePointTrie` implementation.
+
+#### Option 2: Implement a reader for the ICU4C `CodePointTrie` binary data directly in Rust in ICU4X
+
+This option entails writing Rust code that can interpret the binary serialization of the `CodePointTrie` and navigate it directly. It would require also creating an "offline" step (relative to ICU4X) in which ICU4C binary data is exported as a companion package of data in the data downloads for new each ICU release.
+
+The advantages would be having code in ICU4X that shares the same precomputed optimized code point trie data, and it is therefore guaranteed to stay in sync with future improvements in `CodePointTrie` implementation in order to use new versions of data. Only the `CodePointTrie` reader code needs to be ported to read the serialized data, which is much smaller than the C++ code in ICU4C for the builder that builds the structure that gets serialized.
+
+The disadvantages would be similar to what are applicable for other external sources of data in ICU4X that go through a data provider (an extra data dependency that requires an offline step for downloading/exporting during build and/or installation time).
+
+### Conclusion
+
+The preferred approach is option 2, to implement a reader in Rust of the ICU binary format of `CodePointTrie`. The disadvantages are fairly minimal and not different from those of other external sources of data for ICU4X's data provider. However, the upsides of optimal performance, less code to write, and better integration with ICU are each considered to be significant.
+
```

Comment:
suggestion: consider adding a sentence saying that it might make sense to implement the builder in ICU4X at some point in the future, but even if we choose to do that, using the ICU4C version is a good way of bootstrapping our implementation for now.